qiime2/core/cache.py

```diff
@@ -484,23 +762,78 @@ def load(self, key):
         return Result._from_archiver(archiver)
 
     def remove(self, key):
-        """Remove a key from the cache then run garbage collection to remove
-        anything it was referencing and any other loose data
+        """Removes a key from the cache then runs garbage collection to remove
+        anything the removed key was referencing and any other loose data.
+
+        Parameters
+        ----------
+        key : str
+            The key we are removing.
+
+        Examples
+        --------
+        >>> from qiime2.sdk.result import Artifact
+        >>> from qiime2.core.testing.type import IntSequence1
+        >>> test_dir = tempfile.TemporaryDirectory(prefix='qiime2-test-temp-')
+        >>> cache_path = os.path.join(test_dir.name, 'cache')
+        >>> cache = Cache(cache_path)
+        >>> artifact = Artifact.import_data(IntSequence1, [0, 1, 2])
+        >>> saved_artifact = cache.save(artifact, 'key')
+        >>> cache.get_keys() == set(['key'])
+        True
+        >>> cache.remove('key')
+        >>> cache.get_keys() == set()
+        True
+        >>> # Note that the data is still in the cache due to our
+        >>> # saved_artifact causing the process pool to keep a reference to it
+        >>> cache.get_data() == set([str(saved_artifact.uuid)])
+        True
+        >>> del saved_artifact
+        >>> # The data is still there even though the reference is gone because
+        >>> # the cache has not run its own garbage collection yet. For various
+        >>> # reasons, it is not feasible for us to safely garbage collect the
+        >>> # cache when a reference in memory is deleted. Note also that
+        >>> # "artifact" is not backed by the data in the cache, it only lives
+        >>> # in memory, but it does have the same uuid as "saved_artifact."
+        >>> cache.get_data() == set([str(artifact.uuid)])
+        True
+        >>> cache.garbage_collection()
+        >>> # Now it is gone
+        >>> cache.get_data() == set()
+        True
+        >>> test_dir.cleanup()
         """
         os.remove(self.keys / key)
         self.garbage_collection()
 
     def clear_lock(self):
-        """Clears the lock on the cache
-        NOTE: Forcibly removes the lock outside of the locking library's API
+        """Clears the flufl lock on the cache.
```

Comment:
This one probably needs more description, it may be worth referencing it in the GC method as well, just so a user has a more complete picture.

Reply:
I'm not sure I have the complete picture. I don't completely remember what motivated us to make this function. I remember we decided it might be useful to be able to remove this lock file (especially when it was the only lock we had), but I don't remember any particular justification or use case for this.